src/modules/sync/diff/modelElements.ts

```diff
@@ -0,0 +1,328 @@
+    maximum_image_size: optionalHandler(baseHandler),
+    allowed_image_types: optionalHandler(baseHandler),
+    allowed_text_blocks: optionalHandler(makeArrayHandler(b => b, () => [])),
+    maximum_text_length: optionalHandler(makeLeafObjectHandler({})),
+    allowed_table_blocks: optionalHandler(makeArrayHandler(b => b, () => [])),
+    allowed_content_types: optionalHandler(makeArrayHandler(ref => ref.codename, () => [])),
+    allowed_item_link_types: optionalHandler(makeArrayHandler(ref => ref.codename, () => [])),
+    allowed_table_formatting: optionalHandler(makeArrayHandler(f => f, () => [])),
+    allowed_table_text_blocks: optionalHandler(makeArrayHandler(b => b, () => [])),
+  });
+
+export const makeTaxonomyElementHandler = (
+  ctx: CommonPropsHandlersContext,
+): Handler<SyncTaxonomyElement> =>
+  makeObjectHandler({
+    ...makeCommonPropsHandlers(ctx),
+    taxonomy_group: (source, target) =>
+      !source.codename || source.codename !== target.codename
+        ? [{ op: "replace", path: "", value: source, oldValue: target }]
+        : [],
+    term_count_limit: optionalHandler(makeLeafObjectHandler({})),
+    default: makeDefaultReferencesHandler(),
+  });
+
+type HandleLinkedItemsElementContext =
+  & CommonPropsHandlersContext
+  & Readonly<{
+    targetItemCodenames: ReadonlySet<string>;
+  }>;
+
+export const makeLinkedItemsElementHandler = (
+  ctx: HandleLinkedItemsElementContext,
+): Handler<SyncLinkedItemsElement> =>
+  makeObjectHandler({
+    ...makeCommonPropsHandlers(ctx),
+    allowed_content_types: optionalHandler(makeArrayHandler(ref => ref.codename, () => [])),
+    item_count_limit: optionalHandler(makeLeafObjectHandler({})),
+    default: makeDefaultReferencesHandler(ctx.targetItemCodenames),
+  });
+
+export const makeSubpagesElementHandler = (
+  ctx: HandleLinkedItemsElementContext,
+): Handler<SyncSubpagesElement> =>
+  makeObjectHandler({
+    ...makeCommonPropsHandlers(ctx),
+    allowed_content_types: optionalHandler(makeArrayHandler(ref => ref.codename, () => [])),
+    item_count_limit: optionalHandler(makeLeafObjectHandler({})),
+    default: makeDefaultReferencesHandler(ctx.targetItemCodenames),
+  });
+
+export const makeTextElementHandler = (ctx: CommonPropsHandlersContext): Handler<SyncTextElement> =>
+  makeObjectHandler({
+    ...makeCommonPropsHandlers(ctx),
+    maximum_text_length: optionalHandler(makeLeafObjectHandler({})),
+    validation_regex: optionalHandler(makeLeafObjectHandler({})),
+    default: optionalHandler(makeLeafObjectHandler({
+      global: ({ value: source }, { value: target }) => source === target,
+    })),
+  });
+
+export const makeDateTimeElementHandler = (
+  ctx: CommonPropsHandlersContext,
+): Handler<SyncDateTimeElement> =>
+  makeObjectHandler({
+    ...makeCommonPropsHandlers(ctx),
+    default: optionalHandler(makeLeafObjectHandler({
+      global: ({ value: source }, { value: target }) => source === target,
+    })),
```

Comment:
nitpick: again we have here three times in a row some default handler :/ maybe extract this too?